consensus/src/test/kotlin/maru/consensus/state/StateTransitionImplTest.kt

```diff
@@ -65,11 +65,14 @@ class StateTransitionImplTest {
     object : ValidatorProvider {
       override fun getValidatorsForBlock(header: BeaconBlockHeader): SafeFuture<Set<Validator>> =
         SafeFuture.completedFuture(validators)
+
+      override fun getValidatorsForBlock(blockNumber: ULong): SafeFuture<Set<Validator>> =
+        SafeFuture.completedFuture(validators)
     }
 
   private val proposerSelector =
     object : ProposerSelector {
-      override fun getProposerForBlock(header: BeaconBlockHeader): SafeFuture<Validator> =
+      override fun selectProposerForRound(header: BeaconBlockHeader): SafeFuture<Validator> =
```

Comment:
FYI, I changed it to accept `ConsensusRoundIdentifier` instead in another PR. [here](https://github.com/Consensys/maru/pull/63#discussion_r2011676843) is the relevant discussion. 

I see you've added the required `getSealedBeaconBlock(blockNumber: ULong)` to the BeaconChain, so we're aligned here

Reply:
Now I see that you've added another `ProposerSelector` accepting `ConsensusRoundIdentifier`, so this one can be removed now